class_15_sockets/homework/metric.py

```diff
@@ -0,0 +1,37 @@
+import operator
+import threading
+import time
+from itertools import starmap
+
+
+class Metric:
+    def __init__(self, name, func, interval=1):
+        self.name = name
+        self.func = func
+        self.values = []
+        self._running = False
+        self.interval = interval
+        self.t = None
+
+    def start_collect(self):
+        self.t = threading.Thread(name=self.name, target=self._run, daemon=True)
+        self._running = True
+        self.t.start()
+
+    def stop_collect(self):
+        self._running = False
+
+    def _run(self):
+        while self._running:
+            self.values.append(self.func())
+            time.sleep(self.interval)
+
+    def cleanup(self):
+        self.values = []
+
+    def get_current_state(self):
+        return self.name, self.values
+
+
+def count_moves(pos):
```

Comment:
isn't used in code, why?

Reply:
I was thinking on the best way of implementing it and didn't find a good solution to add it into Metric class. So I left class to work with raw data and not doing any aggregation. 
In my understanding aggregation should be done in another kind of abstraction like MerticCollector or something which would get set of pairs: (metric, aggr function) as incoming paramters.
I didn't come up with a solution in a reasonable time but left this code in case I'd like to make it happen in a future.
Don't you mind I keep it in there even though it's not used anywhere?